api/src/main/java/org/apache/iceberg/Scan.java

```diff
@@ -29,7 +29,7 @@
  * Scan objects are immutable and can be shared between threads. Refinement methods, like
  * {@link #select(Collection)} and {@link #filter(Expression)}, create new TableScan instances.
  */
-public interface Scan<T extends Scan<T>> {
+public interface Scan<ThisT, T extends ScanTask, S extends InputSplit<T>> {
```

Comment:
I am debating whether we need to attach a boundary to `ThisT`. It won't harm but we don't do that in any other places and it makes the definition a bit more cumbersome as `Scan` has 3 params now.

Reply:
This is still open.